server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionFinalizer.java

```diff
@@ -100,8 +102,11 @@ protected CompactionFinalizer(ServerContext context, ScheduledThreadPoolExecutor
     ThreadPools.watchCriticalScheduledTask(schedExecutor.scheduleWithFixedDelay(
         this::notifyTservers, 0, tserverCheckInterval, TimeUnit.MILLISECONDS));
 
-    this.sharedBatchWriter =
-        new SharedBatchWriter(Ample.DataLevel.USER.metaTable(), context, queueSize);
+  }
+
+  private SharedBatchWriter getWriter(ExternalCompactionId ecid) {
+    return writers.computeIfAbsent(ecid.canonical().charAt(0),
```

Comment:
External compaction ids have a prefix of `ECID:`, so charAt(0) will probably always be a `E`

Reply:
Forgot about the prefix. Fixed in 3ebccfc